### ISSUE TITLE
Possible refactor to allow more use within Builder

### ISSUE THREAD

**Issue by peircej** [herrlich10/psykit#1]
Hi @herrlich10 Great to see extensions like this. Definitely filling a requested functionality 👏 and hopefully our new plugin system makes them easier to install into PsychoPy by users. 

One suggestion that would make it easier for this to be used within a PsychoPy experiment generated by Builder (the way most users access it) is that it would be great if your `StereoWindow` took a `Window` _instance_ as an argument and then adjusted/augmented that instance rather than being a subclass. The reason this is useful is that it's not easy to replace a line in builder boilerplate whereas it is easy to add one. So the ideal ultimate code would be

```
win = visual.Window(....)
_stereoWindow = psykist.StereoWindow(win, stereoMode='left/right', crossTalk=None)
# win is now adapted for stereo use
```

This system was first designed for, and tested with, the WindowWarper class which allows custom screen shapes to be added to a window. To make it possible we added some hooks to the Window class so that you could easily add to code to the beginning of a flip, or the end of a flip

For instance, rather than overriding `Window.flip()` with your code to set up the viewport, you would insert that code into a function that overrides `Window._startOfFlip` and you would insert your end-of-flip code into `Window._endOfFlip`

I also wonder whether `psykit` is the ideal name for your code to be found:
- there is also a vpixx plugin in development at https://github.com/wichmann-lab/psychopy-pixx and your code might be best placed there if the primary usecase is for propixx displays
- or maybe something dedicated to stereo since it seems this would work beyond vpixx hardware

do email me if this would be worth a discussion on zoom

and thanks again for your efforts supporting the community :-)

**Comment by herrlich10** [comment]
Hi @peircej,

I'm very excited to see your comment here 😃 Apologies for the delayed response.
First, I want to say thank you for initiating and leading the continuous development of Psychopy. It is truly a fantastic tool. I'm delighted to finally be able to give something back after receiving so much help from the community.

Regarding your suggestion about the calling signature for easier integration with the Builder, I have updated the code to support both the adaptor paradigm (`stereoWin = StereoWindow(win, stereoMode='left/right')`, see [demos/minimum_adaptor.py](https://github.com/herrlich10/psykit/blob/master/psykit/demos/minimum_adaptor.py)) and the original subclass initialization (`stereoWin = StereoWindow(stereoMode='left/right')`, see [demos/minimum_example.py](https://github.com/herrlich10/psykit/blob/master/psykit/demos/minimum_example.py)). I kept the subclassing implementation though, to minimize the change of code, and I think the end users should hardly feel any difference. Of course, more tests and tweaks are need to make StereoWindow work with various components in Psychopy. Some interactive features might still have issues, though I haven't had time to test them yet.

As for the name of the package, I was partly inspired by `scikit` (which also sounds similar). I initially intended to use this package as an umbrella for potential future functional modules, with `stereomode` being the first of them. As you mentioned, the `stereomode` module aims to provide various stereo modes (for use with mirrors, prisms, shutter glasses, anaglyph glasses, etc.) and working with propixx is only one use case. Therefore, placing it in [psychopy-pixx](https://github.com/wichmann-lab/psychopy-pixx) might not be ideal. Maybe we can add a pointer on the main PsychoPy API page to direct users who need stereo or dichoptic visual displays to this extension. If necessary, we could wrap it as a plugin after it has been tested by other users (currently, it works well for my own experiments). Your suggestions and insights will be invaluable as we move forward.

Thank you once again for your input!
